community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/birk/il/Instruction.scala

```diff
@@ -28,16 +30,20 @@ trait Instruction {
   //generate class level members
   def fields(): String
 
-  // Initialises necessary data-structures. Is inserted at the top of the generated method
+  //Initialises necessary data-structures. Is inserted at the top of the generated method
   def generateInit(): String
 
   def methods: Seq[Method] = {
     (allLeafs :+ this).flatMap(_._method)
   }
 
+  def exceptions: Set[ExceptionCodeGen] = allLeafs.flatMap(_._exceptions()).toSet
```

Comment:
Having a final method an a trait seems a bit weird. However I do agree that this needs to be cleaned up but let's do that as separate PR.